### PR TITLE
[Concepts]: Add while loops

### LINKED ISSUE #399: [Concepts]: Add while loops
# Goal

The goal of this exercise is to teach the student about while loops in crystal.

## Learning objectives

- Know about while and until statements
- Know how to create a basic while loop
- Know how to skip to the next iteration of a loop
- Know how to exit a loop early

## Out of scope

- For loops

## Concepts

`While-loops`:

- while
- until
- next
- break

## Prerequisites

- Conditionals

### PR DESCRIPTION
**Pull request by meatball133**
@angelikatyborska
@glennj

resolves #399

### CODE REVIEW

File: concepts/while-loops/about.md [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+a = while true
+  break 1
+end
+
+p a
+# => 1
+
+a = while true
+  1
+end
```
Comment (marksiemers) [inline concepts/while-loops/about.md:138]: This looks like in infinite loop

File: concepts/return/introduction.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
+# Return
+
+In Crystal so does always the last expression in a method return its value.
```
Comment (marksiemers) [inline concepts/return/introduction.md:3]: The introduction and about look the same, is this intentional?

**Review by marksiemers — COMMENTED** [review]
This looks pretty good.

I may not understand how a concept is supposed to be added to Exercism, but it looks like a lot of duplicative content (about and introduction).

File: concepts/while-loops/about.md (hunk 1 above)
Comment (meatball133) [reply concepts/while-loops/about.md:138]: Yeah, that is not intended to be that way. Thanks

File: concepts/return/introduction.md (hunk 2 above)
Comment (meatball133) [reply concepts/return/introduction.md:3]: The introduction is about how the concept looks before you start the exercise and about is what it looks after. I quite often add some extra tidbit in about but other than that so are they the same

**Comment by meatball133** [comment]
Hi!

Yes, there is some duplicate-looking content but they are slightly different and each file has a different purpose.

**Review by glennj — CHANGES_REQUESTED** [review]
Just a few documentation suggestions.

The chosen exercise does not really demonstrate much of the concept material (no early return, no break/next). I'm going to look around for an exercise that might be more specifically relevant.

**Review by ErikSchierboom — APPROVED** [review]

**Pull request merged**